src/features/chart/redux/actions.ts

```diff
@@ -0,0 +1,45 @@
+import { UPDATE_CHART_RANGE, UPDATE_CHART_DATA } from "./constants";
+import {
+  ActionCreator,
+  Action,
+  ActionWithPayload
+} from "../../../utils/actions";
+import { RESET } from "../../../store/constants";
+
+export interface ChartData {
+  close: number;
+  date: string;
+}
+
+export type ResetAction = Action<RESET>;
+
+export const updateChartRangeAction: ActionCreator<
+  typeof UPDATE_CHART_RANGE,
+  string
+> = chartRange => ({
+  type: UPDATE_CHART_RANGE,
+  payload: chartRange
+});
+
+export type UpdateChartRangeAction = ActionWithPayload<
+  UPDATE_CHART_RANGE,
+  string
+>;
+
+export type UpdateChartDataAction = ActionWithPayload<
+  UPDATE_CHART_DATA,
+  ChartData[]
+>;
+
+export type ChartActions =
+  | UpdateChartDataAction
+  | UpdateChartRangeAction
+  | ResetAction;
+
+export const updateChartDataAction: ActionCreator<
```

Comment:
Please try to group `type` and `const` declarations together in a cohesive order, so it's easier to read.
Your current `const`-`type` order is:
1. `const updateChartRangeAction`
2. `type UpdateChartRangeAction`
3. `type UpdateChartDataAction`
4. `type ChartActions`
5. `const updateChartDataAction`
which is all a bit muddled.